rear-panel/core/lights/animation_math.cpp

```diff
@@ -0,0 +1,56 @@
+#include "rear-panel/core/lights/animation_math.hpp"
+
+#include <cmath>
+
+#include "rear-panel/core/lights/animation_queue.hpp"
+
+static constexpr double PI = std::acos(-1);
+
+static auto calculate_linear_power(double start_power, double end_power,
+                                   uint32_t ms_count, uint32_t ms_total)
+    -> double {
+    // Simple linear regression. At count of 0ms we return start_power, and at
+    // the end we return the end_power.
+    double slope = end_power - start_power;
+    double scale =
+        static_cast<double>(ms_count) / static_cast<double>(ms_total);
+    return (scale * slope) + start_power;
+}
+
+static auto calculate_sin_power(double start_power, double end_power,
+                                uint32_t ms_count, uint32_t ms_total)
+    -> double {
+    double slope = end_power - start_power;
```

Comment:
nit: `slope` is kind of a bad name for this, it would only be the slope if it was divided by the time. can we call it `powerdiff` or something